File: MCRunner/Orders/MarketOrder.cs
using PowerLanguage;
using System;

namespace MCRunner.Orders
{
    public class MarketOrder : BaseOrder, IOrderMarket
    {
        public MarketOrder(SOrderParameters orderParams, bool openNext = false)
            : base(orderParams, OrderCategory.Market, openNext)
        {
        }

        public void Send()
        {
            TriggerOrderSent(new OrderInfo()
            {
                Order = this,
                OrderAction = OrderParams.Action,
                OrderExit = OrderParams.ExitTypeInfo,
                Size = OrderParams.Lots.GetSize()
            });
        }

        public void Send(int numLots)
        {
            if (numLots < 0)
            {
                throw new ArgumentOutOfRangeException("numLots must be larger than 0");
            }
            else if (numLots == 0)
            {
                Send();
            }
            else
            {
                TriggerOrderSent(new OrderInfo()
                {
                    Order = this,
                    OrderAction = OrderParams.Action,
                    OrderExit = OrderParams.ExitTypeInfo,
                    Size = OrderParams.Lots.GetSize(numLots)
                });
            }
        }

        public void Send(string new_name)
        {
            throw new NotImplementedException();
        }

        public void Send(string new_name, int numLots)
        {
            throw new NotImplementedException();
        }

        public void SendFromEntry(string fromName)
        {
            throw new NotImplementedException();
        }

        public void SendFromEntry(int numLots, string fromName)
        {
            throw new NotImplementedException();
        }

        public void SendFromEntry(string new_name, string fromName)
        {
            throw new NotImplementedException();
        }

        public void SendFromEntry(string new_name, int numLots, string fromName)
        {
            throw new NotImplementedException();
        }
    }
}

File: python/mcrunner/orders/market_order.py
from .enums import SOrderParameters, OrderCategory, EOrderAction
from .order_info import OrderInfo
from .base_order import BaseOrder

class MarketOrder(BaseOrder):
    def __init__(self, order_params: SOrderParameters, open_next: bool = False):
        super().__init__(order_params, OrderCategory.Market, open_next)

    def send(self, num_lots: int = 0):
        if num_lots < 0:
            raise ValueError("numLots must be >= 0")
        size = self.OrderParams.contracts.contract if not self.OrderParams.contracts.is_user_specified else num_lots or self.OrderParams.contracts.contract
        info = OrderInfo(
            Order=self,
            OrderAction=self.OrderParams.action,
            OrderExit=self.OrderParams.exit_type_info,
            Size=size
        )
        self._trigger(info)
